packages/assistify-ai/server/lib/SmartiAdapter.js

```diff
@@ -35,34 +36,47 @@ export class SmartiAdapter {
 	}
 
 	/**
+	 * Returns a Smarti conversation Id for the given roomId.
 	 *
 	 * @param {*} roomId - the room for which the Smarti conversationId shall be retrieved
-	 * @param {*} message - An optional message for detsailed mapping information
 	 */
-	static _getConversationId(roomId, message) {
-		const smartiResponse = RocketChat.models.LivechatExternalMessage.findOneById(roomId);
-		let conversationId;
+	static getConversationId(roomId) {
 
-		// conversation exists for channel?
-		if (smartiResponse && smartiResponse.conversationId) {
-			conversationId = smartiResponse.conversationId;
+		SystemLogger.debug(`Retrieving conversation ID for channel: ${ roomId }`);
+		const m = RocketChat.models.LivechatExternalMessage.findOneById(roomId);
+		if (m && m.conversationId) {
+			return m.conversationId;
 		} else {
 			SystemLogger.debug('Smarti - Trying legacy service to retrieve conversation ID...');
-			const conversation = SmartiProxy.propagateToSmarti(verbs.get,
-				`legacy/rocket.chat?channel_id=${ roomId }`, null, (error) => {
-					// 404 is expected if no mapping exists
-					if (error.response.statusCode === 404) {
-						return null;
+			const conversation = RocketChat.RateLimiter.limitFunction(
+				SmartiProxy.propagateToSmarti, 5, 1000, {
+					userId(userId) {
+						return !RocketChat.authz.hasPermission(userId, 'send-many-messages');
 					}
-				});
+				}
+			)(verbs.get, `legacy/rocket.chat?channel_id=${ roomId }`, null, (error) => {
+				// 404 is expected if no mapping exists
+				if (error.response.statusCode === 404) {
+					return null;
+				}
+			});
+
 			if (conversation && conversation.id) {
-				conversationId = conversation.id;
-				const timestamp = message ? message.ts : Date.now();
-				SmartiAdapter._updateMapping(roomId, conversationId, timestamp);
+				let timestamp = conversation.messages &&
```

Comment:
Essentially, this is the actual change, right? Instead of utilizing the timestamp from the message, you take the timestamp from the conversation stored in Smarti. How can they differ? I'd love to read some comment about the "why we do it like that" in the code